File: core-ws/src/main/java/com/project/ws/domain/dto/Notification.java
package com.project.ws.domain.dto;

import com.project.ws.domain.enums.MessageTypeEnum;
import com.project.ws.domain.enums.NotificationTypeEnum;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Notification {
    private String id;
    private String chatId;
    private String content;
    private String senderId;
    private List<String> receiverIds;
    private String chatName;
    private MessageTypeEnum messageType;
    private NotificationTypeEnum notificationType;
    private List<String> mediaReferencesList;
}

File: whatsapp-clone-ui/src/app/notification.ts
export interface Notification {
  id?: string;
  chatId?: string;
  content?: string;
  senderId?: string;
  receiversId?: string[];
  messageType?: 'TEXT' | 'MEDIA' | 'AUDIO';
  notificationType?: 'SEEN' | 'MESSAGE' | 'MEDIA' | 'AUDIO' | 'USER_CONNECTED' | 'USER_DISCONNECTED';
  chatName?: string;
  mediaReferencesList?: string[];
}
